### DISCUSSION TITLE
Show Information or warnings only

### DISCUSSION THREAD

**Discussion by Chicagoan2016** [MarimerLLC/csla#4009] [Questions] [answered]
I have to show some 'validation' broken rules' messages as information (or warnings).
Since the 'IsValid' works with 'Errors', how should we check if our object has warnings?
Kind Regards

**Comment by hurcane** [comment, marked as answer]
BrokenRulesCollection is the core container of all broken rules for a business object instance. It has properties for ErrorCount, WarningCount, and ErrorCount. You also have a couple useful methods available:
- `public BrokenRule GetFirstMessage(string property, RuleSeverity severity) - good for showing per property`
- `public string[] ToArray(RuleSeverity severity) - good for all properties of a certain severity`

**Reply by Chicagoan2016** [reply]
@hurcane Thank you Sir.
I remember some discussion about overriding SaveASync for this kind of behavior but I would like to check my object before calling the save method and show warnings to the user.

**Reply by rockfordlhotka** [reply]
What CSLA provides is the ability to override `SaveAsync` to _block_ the save after the user has clicked the save button (or whatever) to trigger the save. By default this only checks for errors, not info or warning values.

You can change that default, but the only thing `SaveAsync` can do is throw an exception back to the UI to indicate that the object is invalid.

If you don't want the user to save an object with outstanding warnings, your options are

1. Make the warnings errors
2. Override `SaveAsync` to throw on warnings as well as errors

If you are trying to tell the user about warnings _before they click save_, then that's a UI concern, and the UI can (as @hurcane says) leverage the broken rule collection.

**Comment by Chicagoan2016** [comment]
@hurcane @rockfordlhotka  Thank you :).
I am going with the UI route since after informing users of warnings, they still have the ability to save the object.
Kind Regards
